### DISCUSSION TITLE
How MMKV keeps data up to date

### DISCUSSION THREAD

**Discussion by Ryuu-64** [Tencent/MMKV#1070] [Q&A] [answered]
Why [android_tutorial_cn](https://github.com/Tencent/MMKV/wiki/android_tutorial_cn) says
"The use of MMKV is very simple, all changes take effect immediately, no need to call `sync`, `apply`."

I see the following methods in `'com.tencent:mmkv-static:1.2.16'` artifact are marked as @Deprecated
Is it because something was changed in the later update, making APIs like `sync` and `apply` unnecessary?

```java
public class MMKV implements SharedPreferences, SharedPreferences. Editor {
     /** @deprecated */
     @Deprecated
     public boolean commit() {
         this. sync(true);
         return true;
     }

     /** @deprecated */
     @Deprecated
     public void apply() {
         this. sync(false);
     }
}
```

If so, how does MMKV achieve automatic persistence and ensure that the persisted information is up-to-date if the data is not manually written to the external memory?

**Comment by lingol** [comment]
https://github.com/Tencent/MMKV/wiki/design

**Comment by lingol** [comment, marked as answer]
MMKV doesn't rely on `commit()` or `apply()` from the very beginning.
